### PR TITLE
CAN C generator

### PR DESCRIPTION
**Pull request by ItsNotSoftware**

### CODE REVIEW

**Comment by ItsNotSoftware** [comment]
Closes #103 
Closes #70

File: plugins/fcp_can_c/templates/can_signal_parser.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+#include <stdint.h>
+
+#include "can_frame.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+uint64_t can_decode_signal_as_uint64_t(const CanFrame *msg, uint64_t start,
```
Comment (joajfreitas) [inline plugins/fcp_can_c/templates/can_signal_parser.h:12]: We have to add more comprehensive tests for these files.

We can leave that for another PR.

File: src/fcp/specs/v2.py [hunk 2]
```diff
@@ -100,6 +100,10 @@ def get_matching_extension(self, struct: Struct, protocol: str) -> Maybe[Impl]:
 
         return Nothing()
 
+    def get_matching_extensions(self, protocol: str) -> List[Impl]:
+        """Get impl for corresponding struct with a specific protocol."""
+        return [extension for extension in self.impls]
+
```
Comment (joajfreitas) [inline src/fcp/specs/v2.py:106]: `get_matching_impls`

I've renamed `extension` to  `impl` following the syntax change. Anyway the protocol argument is not being used currently

**Review by joajfreitas — CHANGES_REQUESTED** [review]
See the comments

**Pull request merged**